api/v1alpha1/dataprotectionapplication_types.go

```diff
@@ -357,6 +359,52 @@ type NodeAgentCommonFields struct {
 	PodConfig *PodConfig `json:"podConfig,omitempty"`
 }
 
+type RestorePVC struct {
+	// IgnoreDelayBinding indicates to ignore delay binding the restorePVC when it is in WaitForFirstConsumer mode
+	IgnoreDelayBinding bool `json:"ignoreDelayBinding,omitempty"`
+}
+
+type RuledConfigs struct {
+	// NodeSelector specifies the label selector to match nodes
+	NodeSelector metav1.LabelSelector `json:"nodeSelector"`
+
+	// Number specifies the number value associated to the matched nodes
+	Number int `json:"number"`
+}
+
+// LoadConcurrency is the config for data path load concurrency per node.
+type LoadConcurrency struct {
+	// GlobalConfig specifies the concurrency number to all nodes for which per-node config is not specified
+	GlobalConfig int `json:"globalConfig,omitempty"`
+
+	// PerNodeConfig specifies the concurrency number to nodes matched by rules
+	PerNodeConfig []RuledConfigs `json:"perNodeConfig,omitempty"`
+}
+
+// LoadAffinity is the config for data path load affinity.
+type LoadAffinity struct {
+	// NodeSelector specifies the label selector to match nodes
+	NodeSelector metav1.LabelSelector `json:"nodeSelector"`
+}
+
+type NodeAgentConfigMapSettings struct {
+	// LoadConcurrency is the config for data path load concurrency per node.
+	// +optional
+	LoadConcurrency *LoadConcurrency `json:"loadConcurrency,omitempty"`
+	// LoadAffinity is the config for data path load affinity.
+	// +optional
+	LoadAffinity []*LoadAffinity `json:"loadAffinity,omitempty"`
```

Comment:
I don't think we need a separate field here for LoadAffinity. We already have NodeAgentConfig.PodConfig.NodeSelector, and this value (used to schedule datamover pods) needs to match the node selector for the DaemonSet. It's more error-prone if users must manage two separate fields, so it's better to just use the existing node selector value when creating the ConfigMap

Reply:
@sseago we don't need to put anything to the ConfigMap around NodeSelector right? It will apply to node via https://github.com/openshift/oadp-operator/blob/e8b6efd937f3bc60fae55b21ccc7acd9b3205743/internal/controller/nodeagent.go#L293 or am I missing something and we need to include that value in the ConfigMap as well from that setting?

Reply:
https://github.com/openshift/oadp-operator/blob/51cf09b9b8405c66edd7c176afd1f49a8fb9e73c/internal/controller/nodeagent_test.go#L1395-L1418

The above test makes it simpler to review the resulting CM for the above PodConfig, please also see one test below to see full CM with all other config options.